app/Jobs/ProcessNewVideo.php

```diff
@@ -215,60 +228,84 @@ protected function getVideoDimensions($url)
+    protected function extractImagesfromVideo($path, $duration, $destinationPath)
     {
-        // Cache the video instance.
-        if (!isset($this->ffmpegVideo)) {
-            $this->ffmpegVideo = FFMpeg::create()->open($path);
+        $format = config('thumbnails.format');
+        $maxThumbnails = config('videos.sprites_max_thumbnails');
+        $minThumbnails = config('videos.thumbnail_count');
+        $defaultThumbnailInterval = config('videos.sprites_thumbnail_interval');
+        $durationRounded = floor($duration * 10) / 10;
+        $estimatedThumbnails = $durationRounded / $defaultThumbnailInterval;
+        // Adjust the frame time based on the number of estimated thumbnails
+        $thumbnailInterval = ($estimatedThumbnails > $maxThumbnails) ? $durationRounded / $maxThumbnails
+            : (($estimatedThumbnails < $minThumbnails) ? $durationRounded / $minThumbnails : $defaultThumbnailInterval);
+        $frameRate = 1 / $thumbnailInterval;
+
+        // Leading zeros are important to prevent file sorting afterwards
+        $p = Process::fromShellCommandline("ffmpeg -i '{$path}' -vf fps={$frameRate} {$destinationPath}/%04d.{$format}");
+        $p->run();
+        if ($p->getExitCode() !== 0) {
+            throw new Exception("Process was terminated with code {$p->getExitCode()}");
         }
 
-        $buffer = $this->ffmpegVideo->frame(TimeCode::fromSeconds($time))
-            ->save(null, false, true);
-
-        return VipsImage::thumbnail_buffer($buffer, $width, ['height' => $height])
-            ->writeToBuffer(".{$format}", [
-                'Q' => 85,
-                'strip' => true,
-            ]);
     }
 
-    /**
-     * Get the times at which thumbnails should be sampled.
-     *
-     * @param float $duration Video duration.
-     *
-     * @return array
-     */
-    protected function getThumbnailTimes($duration)
+    public function generateVideoThumbnails($disk, $fragment, $tmpDir)
     {
-        $count = config('videos.thumbnail_count');
 
-        if ($count <= 1) {
-            return [$duration / 2];
-        }
+        // Config for normal thumbs
+        $format = config('thumbnails.format');
+        $thumbCount = config('videos.thumbnail_count');
+        $width = config('thumbnails.width');
+        $height = config('thumbnails.height');
 
-        // Start from 0.5 and stop at $duration - 0.5 because FFMpeg sometimes does not
-        // extract frames from a time code that is equal to 0 or $duration.
-        $step = ($duration - 1) / floatval($count - 1);
-        $start = 0.5;
-        $end = $duration - 0.5;
-        $range = range($start, $end, $step);
+        // Config for sprite thumbs
+        $thumbnailsPerSprite = config('videos.sprites_thumbnails_per_sprite');
+        $thumbnailsPerRow = sqrt($thumbnailsPerSprite);
+        $thumbnailWidth = config('videos.sprites_thumbnail_width');
+        $thumbnailHeight = config('videos.sprites_thumbnail_height');
```

Comment:
I think we could also remove the extra `videos.sprites_thumbnail_width` and height and just use the `thumbnails.width` and height for that.